### PR TITLE
Make evaluation script compatible with tensorflow classifier

### PR DESCRIPTION
**Pull request by akelad**
**Proposed changes**:
- Before, entity predictions were always extracted which isn't possible when only using the tensorflow classifier pipeline because it has no tokenizer. Now we check whether an intent classifier/entity extractor is present before running evaluation on either. (this also means no weird confusion matrix is generated when only evaluating an NER pipeline)
- We also skip evaluation of duckling
- Fixed the crf extractor 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

File: rasa_nlu/evaluate.py [hunk 1]
```diff
@@ -527,17 +516,23 @@ def run_evaluation(data_path, model_path,
...
-    extractors, entity_predictions = patch_duckling(interpreter, extractors,
-                                                    entity_predictions)
 
-    evaluate_intents(intent_targets, intent_predictions)
-    evaluate_entities(entity_targets, entity_predictions, tokens, extractors)
+    if extractors:
+        entity_targets = get_entity_targets(test_data)
+        entity_predictions, tokens = get_entity_predictions(interpreter,
+                                                            test_data)
+        extractors, entity_predictions = remove_duckling(extractors,
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: what happens if only duckling is in the pipeline and we remove it here - will the rest of this if block still work (given that in the beginning we check for `if extractors` which is now empty)?

File: rasa_nlu/evaluate.py [hunk 2]
```diff
@@ -436,6 +436,15 @@ def get_entity_extractors(interpreter):
     return extractors - entity_processors
 
 
+def get_intent_classifier(interpreter):
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: This is never used to actually get the classifiers but only to check if one is present. I'd suggest to call this something like `is_intent_classifier_present` and directly return the bool here

File: rasa_nlu/evaluate.py [hunk 3]
```diff
@@ -638,21 +634,22 @@ def compute_entity_metrics(interpreter, corpus):
     """Computes entity evaluation metrics for a given corpus and
     returns the results
     """
+    entity_results = defaultdict(lambda: defaultdict(list))
+    extractors = get_entity_extractors(interpreter)
+
+    if not extractors:
```
Comment (tmbo) [inline rasa_nlu/evaluate.py]: again, what if duckling is the only extractor present?

File: rasa_nlu/evaluate.py (hunk 1 above)
Comment (akelad) [reply rasa_nlu/evaluate.py]: works fine, yes. Looks like this, although I'm not sure why it's printing `ner_duckling` twice:
```
2018-04-12 13:32:21 INFO     rasa_nlu.model - Starting to train component ner_duckling
2018-04-12 13:32:21 INFO     rasa_nlu.model - Finished training component.
2018-04-12 13:32:22 INFO     __main__ - Skipping evaluation of ner_duckling
2018-04-12 13:32:24 INFO     __main__ - Skipping evaluation of ner_duckling
2018-04-12 13:32:24 INFO     __main__ - CV evaluation (n=2)
2018-04-12 13:32:24 INFO     __main__ - Finished evaluation
```

File: rasa_nlu/evaluate.py (hunk 3 above)
Comment (akelad) [reply rasa_nlu/evaluate.py]: hm wait i see your point here, removing those extractors should happen first - it doesn't break anything, but makes more sense to skip everything else.

**Comment by akelad** [comment]
@tmbo I'd say this is ready to merge now, unless you have any more comments on the changes I made

**Pull request merged**